sdk/src/main/scala/com/horizen/account/wallet/AccountWallet.scala

```diff
@@ -1,53 +1,52 @@
+
 package com.horizen.account.wallet
 
 import com.horizen.account.block.AccountBlock
-import com.horizen.consensus.ConsensusEpochInfo
-import com.horizen.params.NetworkParams
+import com.horizen.node.NodeWalletBase
 import com.horizen.storage.SidechainSecretStorage
-import com.horizen.{SidechainTypes, Wallet}
+import com.horizen.{AbstractWallet, SidechainTypes}
 import scorex.core.VersionTag
+import scorex.util.ScorexLogging
 
 import scala.util.Try
 
-class AccountWallet extends Wallet[SidechainTypes#SCS, SidechainTypes#SCP, SidechainTypes#SCAT, AccountBlock, AccountWallet] {
-  override type NVCT = this.type
-
-  override def addSecret(secret: SidechainTypes#SCS): Try[AccountWallet] = ???
-
-  override def removeSecret(publicImage: SidechainTypes#SCP): Try[AccountWallet] = ???
-
-  override def secret(publicImage: SidechainTypes#SCP): Option[SidechainTypes#SCS] = ???
-
-  override def secrets(): Set[SidechainTypes#SCS] = ???
 
-  override def publicKeys(): Set[SidechainTypes#SCP] = ???
-
-  override def scanOffchain(tx: SidechainTypes#SCAT): AccountWallet = ???
+class AccountWallet private[horizen](seed: Array[Byte],
+                                     secretStorage: SidechainSecretStorage)
+  extends AbstractWallet[
+    SidechainTypes#SCAT,
+    AccountBlock,
+    AccountWallet](seed, secretStorage)
+    with ScorexLogging
+    with NodeWalletBase {
+  override type NVCT = this.type
 
-  override def scanOffchain(txs: Seq[SidechainTypes#SCAT]): AccountWallet = ???
 
-  override def scanPersistent(modifier: AccountBlock): AccountWallet = ???
+  //Nothing to rollback for SecretStorage
+  override def rollback(to: VersionTag): Try[AccountWallet] = Try {
+    this
+  }
 
-  override def rollback(to: VersionTag): Try[AccountWallet] = ???
 }
 
-
-object AccountWallet
-{
+object AccountWallet {
   private[horizen] def restoreWallet(seed: Array[Byte],
-                                     secretStorage: SidechainSecretStorage,
-                                     params: NetworkParams): Option[AccountWallet] = {
+                                     secretStorage: SidechainSecretStorage): Option[AccountWallet] = {
 
-    //TODO: uncomment
-    // Some(new AccountWallet(seed, secretStorage, params))
-    Some(new AccountWallet())
+    if (!secretStorage.isEmpty) {
+      Some(new AccountWallet(seed, secretStorage))
+    } else
+      None
   }
 
   private[horizen] def createGenesisWallet(seed: Array[Byte],
-                                           secretStorage: SidechainSecretStorage,
-                                           params: NetworkParams): Try[AccountWallet] = Try {
-    //TODO: uncomment
-    // new AccountWallet(seed, secretStorage, params)
-    new AccountWallet()
+                                           secretStorage: SidechainSecretStorage
+                                          ): Try[AccountWallet] = Try {
+
+    if (secretStorage.isEmpty) {
+      new AccountWallet(seed, secretStorage)
```

Comment:
secret storage can be empty even for long-running node.
`new AccountWallet(seed, secretStorage)` is enough for this method

Reply:
OK.